src/core/derivation.ts

```diff
@@ -185,4 +197,58 @@ export function untrackedStart() {
 
 export function untrackedEnd(prev: boolean) {
 	globalState.isTracking = prev;
-}
\ No newline at end of file
+}
+
+export function changeDependenciesState(targetState, derivation: IDerivation) {
+	const oldState = derivation.dependenciesState;
+	if (oldState === targetState) return;
+	derivation.dependenciesState = targetState;
+	const obs = derivation.observing;
+	const l = obs.length;
+
+	const m = derivation.__mapid;
+
+	// just because it can be bottleneck
+	if (oldState === 0) {
```

Comment:
this code could benefit from some de-dup. the suggested `move` seems eligable indeed.